roles/content_view_promote/tasks/promote.yml

```diff
@@ -0,0 +1,87 @@
+---
+# get data on the current content-view
+- name: "Gather Data For Current Content-View From Foreman"
+  theforeman.foreman.resource_info:
+    username: "{{ foreman_user }}"
+    password: "{{ foreman_password }}"
+    server_url: "{{ foreman_server_url }}"
+    organization: "{{ organization.key }}"
+    resource: content_views
+    search: name = "{{ content_view }}"
+    validate_certs: false
+  register: content_view_data
+
+# get data on the current content-view version
+- name: "Gather Data For Current Content-View Versions From Foreman"
+  theforeman.foreman.resource_info:
+    username: "{{ foreman_user }}"
+    password: "{{ foreman_password }}"
+    server_url: "{{ foreman_server_url }}"
+    organization: "{{ organization.key }}"
+    resource: content_view_versions
+    params:
+      content_view_id: "{{ content_view_data.resources[0].id }}"
+  register: version_information
+
+# creates a dictionary with data formatted as such {'Prod':'11.0'}
+- name: "Build Dictionary With Lifecycle Envrionment And Version Number"
+  set_fact:
+    environments: "{{ environments | default({}) | combine ({item[1].name : item[0].name.split()[-1]}) }}"
+  with_subelements:
+    - "{{ version_information.resources }}"
+    - environments
+
+# create list of content-view versions
+- name: "Build List of All Versions of Content-View"
+  set_fact:
+    cv_versions: "{{ cv_versions | default([]) + [item.major] }}"
+  with_items: "{{ version_information.resources }}"
+
+# set the highest version to zero so that we don't use previous Content-View settings
+- name: "Set Highest Version to 0"
+  set_fact:
+    highest_version: 0
+
+# set highest number
+- name: "Set the Highest Version of the Content-View Currently Available"
+  set_fact: 
+    highest_version: "{{ cv_versions | max }}"
+
+# add one to each of the version numbers
+- name: "Update Facts With Incremented Content-View Version Numbers"
+  set_fact:
+    new_environments: "{{ new_environments | default({}) | combine ({item.key: item.value|int + 1.0 }) }} "
+  with_dict: "{{ environments }}"
+
+- name: "Check if publishing new version of Content-View is necessary"
+  set_fact:
+    publish_true: true
+  with_dict: "{{ new_environments }}"
+  when: item.value | int > highest_version | int
+
+# only publish new view if necessary (when the current view is at or above library)
+- name: "Publish new version of Content-View"
```

Comment:
I think you can drop this as I wouldn't expect a promote role to publish as well.

Reply:
If the LCE is already at the latest version of a CV then there is nowhere to promote it to.

Are you suggesting that we should have this fail as an alternative?

I can do that, it just seems a bit hostile, especially when its trivial to accommodate this way.

Reply:
we're working on this in https://github.com/theforeman/foreman-ansible-modules/pull/1186 :)